Jenkinsfile

```diff
@@ -1,28 +1,39 @@
-#!groovy
+#!/usr/bin/env groovy
 
 /* Only keep the 10 most recent builds. */
 properties([[$class: 'BuildDiscarderProperty',
                 strategy: [$class: 'LogRotator', numToKeepStr: '10']]])
 
-node {
-  stage 'Checkout'
-  checkout scm
-
-  stage 'Build'
-
-  /* Call the maven build. */
-  mvn "clean install -B -V -U -e -Dsurefire.useFile=false -Dmaven.test.failure.ignore=true"
-
-  /* Save Results. */
-  stage 'Results'
-
-  /* Archive the test results */
-  step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])
-
-  /* Archive the build artifacts */
-  step([$class: 'ArtifactArchiver', artifacts: 'target/*.hpi,target/*.jpi'])
+List labels = ['linux', 'windows']
```

Comment:
@MarkEWaite I haven't found a good way to detect whether there's a label available unfortunately. In the case of dynamically provisioned agents I'm not actually sure how this would even work :frowning_man: